google/cloud/spanner_dbapi/connection.py

```diff
@@ -121,6 +127,31 @@ def instance(self):
         """
         return self._instance
 
+    @property
+    def read_only(self):
+        """Flag: the connection can be used only for database reads.
+
+        Returns:
+            bool:
+                True, if the connection intended to be used for
+                database reads only.
+        """
+        return self._read_only
+
+    @read_only.setter
+    def read_only(self, value):
+        """`read_only` flag setter.
+
+        Args:
+            value (bool): True for ReadOnly mode, False for ReadWrite.
+        """
+        if self.inside_transaction:
+            raise ValueError(
+                "Connection read/write mode can't be changed while a transaction is in progress. "
+                "Commit or rollback the current transaction and try again."
+            )
+        self._read_only = value
```

Comment:
It's done in the same way JDBC works: connection mode can be changed, but only if there is no active transaction on it.